libs/designsystem/header-actions/src/header-actions.component.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  HostBinding,
+  Input,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ButtonComponent } from '@kirbydesign/designsystem/button';
+import { DropdownComponent, DropdownModule } from '@kirbydesign/designsystem/dropdown';
+
+@Component({
+  selector: 'kirby-header-actions',
+  standalone: true,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [CommonModule, DropdownModule],
+  templateUrl: './header-actions.component.html',
+  styleUrls: ['./header-actions.component.scss'],
+})
+export class HeaderActionsComponent implements AfterViewInit {
+  /*
+   * When the consumer sets the visibleActions input, we run all our initialization logic.
+   * The guard for this.buttons exist because when the component is instantiated,
+   * the ContentChild query has not yet run, and all the below functions assume that
+   * the buttons array is defined. When undefined, we defer the initialization to ngAfterViewInit().
+   */
+  @Input() set visibleActions(value: number) {
+    this._visibleActions = value;
+
+    if (!this.buttons) return;
+    this.initializeCollapsing();
+  }
+
+  @HostBinding('class')
+  @Input()
+  placement: 'left' | 'right' = 'left';
+
+  @ViewChild(DropdownComponent, { read: ElementRef }) dropdown!: ElementRef;
+  @ContentChildren(ButtonComponent, { read: ElementRef }) buttons!: ElementRef<HTMLButtonElement>[];
+  @ViewChild('hiddenLayer', { read: ElementRef }) hiddenLayer!: ElementRef<HTMLElement>;
+  @ViewChild('visibleLayer', { read: ElementRef }) visibleLayer!: ElementRef<HTMLElement>;
+
+  _collapsedActions: string[] = [];
+  _visibleActions: number;
+
+  private dropdownTextToButtonMap: Map<string, HTMLButtonElement>;
+
+  constructor(private renderer: Renderer2) {}
+
+  ngAfterViewInit(): void {
+    this.initializeCollapsing();
+  }
+
+  initializeCollapsing() {
+    if (this.buttons.length <= 2) return;
+    this.moveButtons();
+    this.populateDropdown();
+    this.toggleDropdown();
+  }
+
+  moveButtons() {
+    this.buttons.forEach((button, index) => {
+      if (index > this._visibleActions - 1) {
+        this.renderer.appendChild(this.hiddenLayer.nativeElement, button.nativeElement);
+      } else {
+        this.renderer.appendChild(this.visibleLayer.nativeElement, button.nativeElement);
+      }
+    });
+  }
+
+  toggleDropdown() {
+    if (this.hiddenLayer.nativeElement.childElementCount === 0) {
+      this.renderer.setStyle(this.dropdown.nativeElement, 'display', 'none');
+    } else {
+      this.renderer.setStyle(this.dropdown.nativeElement, 'display', 'inline-block');
+    }
```

Comment:
Consider if this could be simplified by:
- setting a prop with a hostbinding to `class.isCollapsed` or similar
- styling the dropdown with `host('.isCollapsed´) kirby-dropdown { display: inline-block; }`